orchestrator/careplancontributor/ehr/kafka_client.go

```diff
@@ -73,52 +87,68 @@ type NoopClientImpl struct {
 //   - KafkaClient: An implementation of the KafkaClient interface.
 //   - error: An error if the Kafka producer could not be created.
 func NewClient(config KafkaConfig) (KafkaClient, error) {
-
 	var kafkaClient KafkaClient
 	if config.Enabled {
-		endpoint := config.Endpoint
-		log.Infof("KafkaClientImpl, connecting to %s, Mechanism: %s, protocol: %s, username: %s, password: %s",
-			endpoint,
-			config.Sasl.Mechanism,
-			config.Security.Protocol,
-			config.Sasl.Username,
-			config.Sasl.Password)
-		producer, err := kafka.NewProducer(&kafka.ConfigMap{
-			"bootstrap.servers": endpoint,
-			"sasl.mechanisms":   config.Sasl.Mechanism,
-			"sasl.username":     config.Sasl.Username,
-			"sasl.password":     config.Sasl.Password,
-			"security.protocol": config.Security.Protocol,
-		})
-		if err != nil {
+		if config.DebugOnly {
+			ctx := context.Background()
+			log.Info().Ctx(ctx).Msg("Debug mode enabled, writing messages to files in /tmp")
```

Comment:
Temp dir is platform dependant
```suggestion
			log.Info().Ctx(ctx).Msg("Debug mode enabled, writing messages to files in OS temp dir")
```